client/src/main/java/io/split/client/LocalhostSplitFactory.java

```diff
@@ -19,26 +20,36 @@
     private static final Logger _log = LoggerFactory.getLogger(LocalhostSplitFactory.class);
 
     static final String FILENAME = ".split";
+    static final String FILENAME_YAML = ".split.yaml";
     static final String LOCALHOST = "localhost";
 
     private final LocalhostSplitClientAndFactory _client;
     private final LocalhostSplitManager _manager;
-    private final LocalhostSplitFile _splitFile;
+    private final AbstractLocalhostSplitFile _splitFile;
 
     public static LocalhostSplitFactory createLocalhostSplitFactory() throws IOException {
         String directory = System.getProperty("user.home");
         Preconditions.checkNotNull(directory, "Property user.home should be set when using environment: " + LOCALHOST);
         return new LocalhostSplitFactory(directory);
     }
 
-    public LocalhostSplitFactory(String directory) throws IOException {
-        Preconditions.checkNotNull(directory, "directory must not be null");
+    public LocalhostSplitFactory(String directory, String file) throws IOException {
 
         _log.info("home = " + directory);
 
-        _splitFile = new LocalhostSplitFile(this, directory, FILENAME);
+        if (directory == null) {
+            directory = "";
+        }
+
+        File yaml = new File(directory, file);
+
+        if (yaml.exists()) {
```

Comment:
Should we test if the file is readable (permissions-wise) and log an error if it is not?